hummingbot/connector/exchange/ndax/ndax_exchange.py

```diff
@@ -310,8 +570,75 @@ def stop_tracking_order(self, order_id: str):
             del self._in_flight_orders[order_id]
 
     async def _update_order_status(self):
+        """
+        Calls REST API to get order status
+        """
         # Waiting on buy and sell functionality.
-        pass
+        tasks = []
+        if len(self._in_flight_orders) < 1:
+            return
+
+        tracked_orders: List[NdaxInFlightOrder] = list(self._in_flight_orders.values())
+        for tracked_order in tracked_orders:
+            ex_order_id = await tracked_order.get_exchange_order_id()
+            if not ex_order_id:
+                self.logger().debug(f"Tracker order {tracked_order.client_order_id} does not have an exchange id. Attempting fetch in next polling interval")
+                continue
+
+            query_params = {
+                "OmsId": 0,
+                "AccountId": self.account_id,
+                "OrderId": ex_order_id,
+            }
+
+            tasks.append(self._api_request(method="GET",
```

Comment:
It might be a good idea to create an asyncio Task, so the request start processing in the moment. Then the gather can be done without problems for the list of Task objects

Reply:
Yea that makes more sense. Will make the necessary changes